Makefile

```diff
@@ -66,8 +66,8 @@ $(BINDIR)/regexMain.o : $(REGEXDIR)/regexMain.c $(BINDIR)/.keep $(SLJIT_HEADERS)
 $(BINDIR)/regexJIT.o : $(REGEXDIR)/regexJIT.c $(BINDIR)/.keep $(SLJIT_HEADERS) $(REGEXDIR)/regexJIT.h
 	$(CC) $(CPPFLAGS) $(REGEX_CFLAGS) -c -o $@ $(REGEXDIR)/regexJIT.c
 
-$(BINDIR)/sljit_test: $(BINDIR)/.keep $(BINDIR)/sljitMain.o $(BINDIR)/sljitTest.o $(BINDIR)/sljitLir.o
-	$(CC) $(CFLAGS) $(LDFLAGS) $(BINDIR)/sljitMain.o $(BINDIR)/sljitTest.o $(BINDIR)/sljitLir.o -o $@ -lm -lpthread
+$(BINDIR)/sljit_test: $(BINDIR)/.keep $(BINDIR)/sljitMain.o $(TESTDIR)/sljitTest.c $(SRCDIR)/sljitLir.c
```

Comment:
technically we could remove at this point the rule for $(BINDIR)/sljitTest.o that is no longer being used, but it allows for easily testing the broken binary by doing:

```
$ make bin/sljitTest.o
$ cc -DSLJIT_WX_EXECUTABLE_ALLOCATOR -DSLJIT_CONFIG_AUTO=1 -Isljit_src -O2 -Wall  bin/sljitMain.o bin/sljitTest.o bin/sljitLir.o -o bin/broken_test
```

it also makes sure it can build even if there is no SLJIT_MALLOC_EXEC definition even if it is not ideal.